atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/utilities/EteExtension.java

```diff
@@ -0,0 +1,305 @@
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.palantir.atlasdb.AtlasDbEteServer;
+import com.palantir.atlasdb.containers.CassandraEnvironment;
+import com.palantir.atlasdb.ete.GradleExtension;
+import com.palantir.atlasdb.ete.suites.MultiClientWithPostgresTimelockAndPostgresTestSuite;
+import com.palantir.atlasdb.ete.suites.MultiClientWithTimelockAndCassandraTestSuite;
+import com.palantir.atlasdb.ete.suites.SingleClientWithEmbeddedAndCassandraTestSuite;
+import com.palantir.atlasdb.ete.suites.SingleClientWithEmbeddedAndOracleTestSuite;
+import com.palantir.atlasdb.ete.suites.SingleClientWithEmbeddedAndPostgresTestSuite;
+import com.palantir.atlasdb.ete.suites.SingleClientWithEmbeddedAndThreeNodeCassandraTestSuite;
+import com.palantir.atlasdb.http.AtlasDbHttpClients;
+import com.palantir.atlasdb.http.TestProxyUtils;
+import com.palantir.atlasdb.todo.TodoResource;
+import com.palantir.conjure.java.api.config.ssl.SslConfiguration;
+import com.palantir.conjure.java.config.ssl.SslSocketFactories;
+import com.palantir.conjure.java.config.ssl.TrustContext;
+import com.palantir.docker.compose.DockerComposeExtension;
+import com.palantir.docker.compose.connection.Container;
+import com.palantir.docker.compose.connection.DockerMachine;
+import com.palantir.docker.compose.execution.DockerComposeExecArgument;
+import com.palantir.docker.compose.execution.DockerComposeExecOption;
+import com.palantir.docker.compose.logging.LogDirectory;
+import com.palantir.docker.proxy.DockerProxyExtension;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public final class EteExtension implements BeforeAllCallback, ExtensionContext.Store.CloseableResource {
+    private static final GradleExtension GRADLE_PREPARE_TASK =
+            GradleExtension.ensureTaskHasRun(":atlasdb-ete-tests:prepareForEteTests");
+    private static final GradleExtension TIMELOCK_TASK =
+            GradleExtension.ensureTaskHasRun(":timelock-server-distribution:dockerTag");
+    private static final SslConfiguration SSL_CONFIGURATION =
+            SslConfiguration.of(Paths.get("var/security/trustStore.jks"));
+    private static final TrustContext TRUST_CONTEXT = SslSocketFactories.createTrustContext(SSL_CONFIGURATION);
+    private static final short SERVER_PORT = 3828;
+
+    private static final List<Extension> extensions = new ArrayList<>();
+
+    private static DockerComposeExtension docker;
+    private static Duration waitDuration;
+    private static List<String> availableClients;
+
+    private static volatile EteExtension instance;
```

Comment:
This class is actually fully static but to register it as an extension we need an instance. Let's make this instance singleton.